app/src/org/commcare/utils/AndroidArrayDataSource.java

```diff
@@ -18,12 +20,11 @@ public AndroidArrayDataSource(Context context){
 
     @Override
     public String[] getArray(String key) {
-        int resourceId = context.getResources().getIdentifier(key, "array", context.getPackageName());
-
-        if (resourceId == 0) {
-            throw new RuntimeException("Localized array data for '" + key + "' not found");
+        if(key.contains("ethiopian")){
```

Comment:
The imprecision of this worries me. For instance, what if we have 2 ethiopian related arrays: days and months?